hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -399,11 +406,15 @@ class Setting(object):
 
 settings.define_setting(
     'max_iterations',
-    default=1000,
+    default=not_set,
     description="""
 Once this many iterations of the example loop have run, including ones which
```

Comment:
Musing rather than necessarily request for change, but I feel like when we've completely disabled the behaviour of the setting it might be worth updating the description to say something like "This doesn't actually do anything but remains for compatibility reasons".